packages/mui-system/src/createStyled.js

```diff
@@ -90,6 +105,32 @@ function defaultOverridesResolver(slot) {
   return (props, styles) => styles[slot];
 }
 
+const muiStyledFunctionResolver = ({ styledArg, props, defaultTheme, themeId }) => {
+  const resolvedStyles = styledArg({
+    ...props,
+    theme: resolveTheme({ ...props, defaultTheme, themeId }),
+  });
+
+  let optionalVariants;
+  if (isPlainObject(resolvedStyles)) {
+    if (resolvedStyles && resolvedStyles.variants) {
+      optionalVariants = resolvedStyles.variants;
+    }
+    delete resolvedStyles.variants;
+  }
```

Comment:
Is this simpler?

```suggestion
    if (resolvedStyles && resolvedStyles.variants) {
      optionalVariants = resolvedStyles.variants;
      delete resolvedStyles.variants;
    }
```

Reply:
Yeah, it's simpler. I started with simple use-cases of when I need to check for variants & plain object was the first occurrence :) Let me update 👍